### PR TITLE
chore: Deprecating ENABLE_FLASK_COMPRESS

### PR DESCRIPTION
**Pull request by john-bodley**
### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

This PR reverts the logic in https://github.com/apache/incubator-superset/pull/4617 as the Flask-Compress extension includes a `COMPRESS_REGISTER` option (default is `True`) which enables or disables compression when the extension is initialized.

### TEST PLAN

CI.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: docs/installation.rst [hunk 1]
```diff
@@ -228,7 +228,7 @@ Note that the development web
 server (`superset run` or `flask run`) is not intended for production use.
 
 If not using gunicorn, you may want to disable the use of flask-compress
-by setting `ENABLE_FLASK_COMPRESS = False` in your `superset_config.py`
+by setting `COMPRESS_REGISTER = False` in your `superset_config.py`
```
Comment (bkyryliuk) [inline docs/installation.rst:231]: maybe add COMPRESS_REGISTER to the config.py
I tend to use the config.py as a documentation for the available configuration options.

**Review by bkyryliuk — APPROVED** [review]
LG%nit

File: docs/installation.rst (hunk 1 above)
Comment (john-bodley) [reply docs/installation.rst:231]: @bkyryliuk historically `config.py` hasn't enumerated all the various Flask configs but rather only defines those which differ from the defaults.

**Review by villebro — APPROVED** [review]
LGTM

**Comment by mistercrunch** [comment]
LGTM. 

More generally, I'm wondering if it makes sense to avoid doing too much on the Flask app config and instructing people to use `FLASK_APP_MUTATOR` instead. It gets tricky to surface all the knobs and buttons that the ecosystem offers...

I'm pro having good default, but generally against having the burden of resurfacing / managing all those options...

**Pull request merged**